From f35edca465d150a5529b243527d204cbda025cf5 Mon Sep 17 00:00:00 2001
From: Bruno Michel <bmichel@menfin.info>
Date: Thu, 11 Nov 2010 20:05:38 +0100
Subject: [PATCH] Extract the inline style on the authenticity_token wrapper

Reasons:
 * Some people do not like tag-soup
 * Inline CSS is bad
 * Its hard to use display: none instead of display: inline
---
 actionpack/lib/action_view/helpers/form_helper.rb  |    6 +++---
 .../lib/action_view/helpers/form_tag_helper.rb     |    2 +-
 actionpack/test/template/form_helper_test.rb       |    2 +-
 actionpack/test/template/form_tag_helper_test.rb   |    2 +-
 railties/guides/source/form_helpers.textile        |    4 ++--
 .../rails/stylesheets/templates/scaffold.css       |    6 ++++++
 6 files changed, 14 insertions(+), 8 deletions(-)

diff --git a/actionpack/lib/action_view/helpers/form_helper.rb b/actionpack/lib/action_view/helpers/form_helper.rb
index 5cea9ea..f9ac208 100644
--- a/actionpack/lib/action_view/helpers/form_helper.rb
+++ b/actionpack/lib/action_view/helpers/form_helper.rb
@@ -37,7 +37,7 @@ module ActionView
     # The HTML generated for this would be (modulus formatting):
     #
     #   <form action="/people" class="new_person" id="new_person" method="post">
-    #     <div style="margin:0;padding:0;display:inline">
+    #     <div class="authenticity_token_wrapper">
     #       <input name="authenticity_token" type="hidden" value="NrOp5bsjoLRuK8IW5+dQEYjKGUJDe7TQoZVvq95Wteg=" />
     #     </div>
     #     <label for="person_first_name">First name</label>:
@@ -67,7 +67,7 @@ module ActionView
     # the code above as is would yield instead:
     #
     #   <form action="/people/256" class="edit_person" id="edit_person_256" method="post">
-    #     <div style="margin:0;padding:0;display:inline">
+    #     <div class="authenticity_token_wrapper">
     #       <input name="_method" type="hidden" value="put" />
     #       <input name="authenticity_token" type="hidden" value="NrOp5bsjoLRuK8IW5+dQEYjKGUJDe7TQoZVvq95Wteg=" />
     #     </div>
@@ -252,7 +252,7 @@ module ActionView
       # The HTML generated for this would be:
       #
       #   <form action='http://www.example.com' method='post' data-remote='true'>
-      #     <div style='margin:0;padding:0;display:inline'>
+      #     <div class="authenticity_token_wrapper">
       #       <input name='_method' type='hidden' value='put' />
       #     </div>
       #     ...
diff --git a/actionpack/lib/action_view/helpers/form_tag_helper.rb b/actionpack/lib/action_view/helpers/form_tag_helper.rb
index 43ffadc..4018898 100644
--- a/actionpack/lib/action_view/helpers/form_tag_helper.rb
+++ b/actionpack/lib/action_view/helpers/form_tag_helper.rb
@@ -555,7 +555,7 @@ module ActionView
           end
 
           tags = snowman_tag << method_tag
-          content_tag(:div, tags, :style => 'margin:0;padding:0;display:inline')
+          content_tag(:div, tags, :class => 'authenticity_token_wrapper')
         end
 
         def form_tag_html(html_options)
diff --git a/actionpack/test/template/form_helper_test.rb b/actionpack/test/template/form_helper_test.rb
index 76417f4..cf98111 100644
--- a/actionpack/test/template/form_helper_test.rb
+++ b/actionpack/test/template/form_helper_test.rb
@@ -1542,7 +1542,7 @@ class FormHelperTest < ActionView::TestCase
   end
 
   def snowman(method = nil)
-    txt =  %{<div style="margin:0;padding:0;display:inline">}
+    txt =  %{<div class="authenticity_token_wrapper">}
     txt << %{<input name="utf8" type="hidden" value="&#x2713;" />}
     txt << %{<input name="_method" type="hidden" value="#{method}" />} if method
     txt << %{</div>}
diff --git a/actionpack/test/template/form_tag_helper_test.rb b/actionpack/test/template/form_tag_helper_test.rb
index 532f086..f259a62 100644
--- a/actionpack/test/template/form_tag_helper_test.rb
+++ b/actionpack/test/template/form_tag_helper_test.rb
@@ -11,7 +11,7 @@ class FormTagHelperTest < ActionView::TestCase
   def snowman(options = {})
     method = options[:method]
 
-    txt =  %{<div style="margin:0;padding:0;display:inline">}
+    txt =  %{<div class="authenticity_token_wrapper">}
     txt << %{<input name="utf8" type="hidden" value="&#x2713;" />}
     txt << %{<input name="_method" type="hidden" value="#{method}" />} if method
     txt << %{</div>}
diff --git a/railties/guides/source/form_helpers.textile b/railties/guides/source/form_helpers.textile
index ab7301a..d4fb7c6 100644
--- a/railties/guides/source/form_helpers.textile
+++ b/railties/guides/source/form_helpers.textile
@@ -32,7 +32,7 @@ Sample output from +form_tag+:
 
 <html>
 <form action="/home/index" method="post">
-  <div style="margin:0;padding:0">
+  <div class="authenticity_token_wrapper">
     <input name="authenticity_token" type="hidden" value="f755bb0ed134b76c432144748a6d4b7a7ddf2b71" />
   </div>
   Form contents
@@ -341,7 +341,7 @@ output:
 
 <html>
 <form action="/search" method="post">
-  <div style="margin:0;padding:0">
+  <div class="authenticity_token_wrapper">
     <input name="_method" type="hidden" value="put" />
     <input name="authenticity_token" type="hidden" value="f755bb0ed134b76c432144748a6d4b7a7ddf2b71" />
   </div>
diff --git a/railties/lib/rails/generators/rails/stylesheets/templates/scaffold.css b/railties/lib/rails/generators/rails/stylesheets/templates/scaffold.css
index 1ae7000..bc81e7f 100644
--- a/railties/lib/rails/generators/rails/stylesheets/templates/scaffold.css
+++ b/railties/lib/rails/generators/rails/stylesheets/templates/scaffold.css
@@ -54,3 +54,9 @@ div.field, div.actions {
   font-size: 12px;
   list-style: square;
 }
+
+.authenticity_token_wrapper {
+  margin: 0;
+  padding: 0;
+  display: inline;
+}
-- 
1.7.1

